src/javascripts/application.mjs

```diff
@@ -56,9 +57,14 @@ $tabs.forEach(($tabs) => {
 // Do this after initialising tabs
 new OptionsTable()
 
+// Attach some responsive code to code blocks
+new CodeBlocks(document.querySelectorAll('[data-module="app-code-block"]'))
```

Comment:
With the work we're doing on component initialisation, it'd be neat if this component ran the same way as the rest of the components by first doing the `document.querySelectorAll` and then initialising an individual component per element found. This will save some refactoring when [using `createAll` to initialise the components of the Design System site](https://github.com/alphagov/govuk-design-system/issues/3812).

If the reason for doing the looping inside the component was to share the ResizeObserver, it's fine to use a `let` variable in the component's module file or a static property on the component.

Reply:
The intent was so it could use the same ResizeObserver for all of the code blocks, yeah. I can rework it to do that differently.